src/storage/documentService/documentService.js

```diff
@@ -1,43 +1,64 @@
-const verify = require("@govtechsg/oa-verify");
+const { verify } = require("@govtechsg/oa-verify");
 const uuid = require("uuid/v4");
-const { encryptString } = require("./crypto");
+const {
+  encryptString,
+  generateEncryptionKey
+} = require("@govtechsg/opencerts-encryption");
 const config = require("../config");
 const { put, get, remove } = require("../dynamoDb");
 
 const DEFAULT_TTL = 60 * 60; // 1 Hour
 const MAX_TTL = 60 * 60 * 24 * 30; // 30 Days
 
-const putDocument = async (document, ttl = DEFAULT_TTL) => {
+const putDocument = async (
+  document,
+  id,
+  ttl = DEFAULT_TTL,
+  conditionalParams
+) => {
   // TTL is handled by dynamoDb natively, this timestamp has to be UTC unixtime in seconds
   const created = Math.floor(Date.now() / 1000);
-  // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/time-to-live-ttl-how-to.html
   const params = {
     TableName: config.dynamodb.storageTableName,
     Item: {
-      id: uuid(),
+      id,
       document,
       created,
       ttl: created + ttl
-    }
+    },
+    ...conditionalParams
   };
-
   return put(params).then(() => params.Item);
 };
 
-const getDocument = async (id, { cleanup }) => {
+const getDocument = async (id, { cleanup } = { cleanup: false }) => {
   const params = {
-    TableName: process.env.OA_DOC_STORAGE_TABLE,
+    TableName: config.dynamodb.storageTableName,
     Key: {
       id
     }
   };
   const document = await get(params);
+  if (document.awaitingUpload) {
+    throw new Error("No Document Found");
```

Comment:
```suggestion
    // we throw this error because if awaitingUpload exists on an object, it also has a decryption key in it and we don't want to return that, ever
    throw new Error("No Document Found");
```

Reply:
added

Reply:
added